src/NodaTime/TimeInterval.cs

```diff
@@ -0,0 +1,284 @@
+        /// <value><c>true</c> if this TimeInterval has a fixed end point, or <c>false</c> if it
+        /// extends to the end of time.</value>
+        public bool HasEnd => end.HasValue;
+
+
+        /// <summary>
+        /// Returns <c>true</c> if Start of TimeInterval is greater than End (for intervals like 23:00/1:00)
+        /// </summary>
+        public bool IsEndInNextDay { get; }
+
+        /// <summary>
+        /// Returns the Period of the TimeInterval.
+        /// </summary>
+        /// <remarks>
+        /// This will always be a non-negative Period, though it may be zero.
+        /// </remarks>
+        /// <value>The Period of the TimeInterval.</value>
+        /// <exception cref="InvalidOperationException">The TimeInterval extends to the start or end of time.</exception>
+        public Period Period
+        {
+            get
+            {
+                var duration = Duration;
+                return new Period(0, 0, 0, 0, 
+                    duration.Hours, duration.Minutes, duration.Seconds, duration.Milliseconds, 0, 
+                    duration.NanosecondOfDay % NodaConstants.NanosecondsPerMillisecond);
+            }
+        }
+
+        /// <summary>
+        /// Returns the Duration of the TimeInterval.
+        /// </summary>
+        /// <remarks>
+        /// This will always be a non-negative Duration, though it may be zero.
+        /// </remarks>
+        /// <value>The Duration of the TimeInterval.</value>
+        /// <exception cref="InvalidOperationException">The TimeInterval extends to the start or end of time.</exception>
+        public Duration Duration
+        {
+            get
+            {
+                var nanoseconds = End.NanosecondOfDay - Start.NanosecondOfDay;
+                if (IsEndInNextDay) nanoseconds += NodaConstants.NanosecondsPerDay;
+                return Duration.FromNanoseconds(nanoseconds);
+            }
+        }
+
+        /// <summary>
+        /// Returns whether or not this TimeInterval contains the given LocalTime.
+        /// </summary>
+        /// <param name="check">LocalTime to test.</param>
+        /// <returns>True if this TimeInterval contains the given LocalTime; false otherwise.</returns>
+        [Pure]
+        public bool Contains(LocalTime check)
+        {
+            if (IsEndInNextDay)
+            {
+                if (check < Start) return check < End;
+                return true;
+            }
+            return check >= Start && check < End;
+        }
+
+        /// <summary>
+        /// Deconstruct this value into its components.
+        /// </summary>
+        /// <param name="_start">The start of the TimeInterval.</param>
+        /// <param name="_end">The end of the TimeInterval.</param>
+        [Pure]
+        public void Deconstruct(out LocalTime? _start, out LocalTime? _end)
+        {
+            _start = start;
+            _end = end;
+        }
+
+        #region Implementation of IEquatable<TimeInterval>
```

Comment:
Let's get rid of all these regions before they even appear. They're *somewhat* helpful for more complex types, but I don't think there's value here. There are plenty I should probably get rid of in the existing code but haven't got round to...

Reply:
So, just delete region IEquatable, its interface inheritance and move Equals logic into object overridden method?

Reply:
I'm just suggesting removing all of the `#region` and `#endregion` lines - no need for anything to be moved or removed.

Reply:
Regions removed.